src/irmin-pack/IO.ml

```diff
@@ -238,15 +269,22 @@ module Unix : S = struct
           match actual_version with
           | `V1 ->
               Log.debug (fun l -> l "[%s] file exists in V1" file);
-              let offset = Raw.Offset.get raw in
-              v ~offset ~version:`V1 ~generation:Int63.zero raw
+              let offset =
+                if readonly then Int63.zero else Raw.Offset.get raw
```

Comment:
this is a bit odd, why is this 0 in the read-only case?

Reply:
ha it seems to be documented bellow - but I am not sure to understand the difference of behaviour between read-only and read-write here.